keps/0001a-meta-kep-implementation.md

```diff
@@ -0,0 +1,206 @@
+
+### Non-Goals
+
+- API Review process
+- Feature request triage
+- Developer guide
+
+## Proposal
+
+### Implementation Details / Notes / Constraints
+
+#### Define
+
+- Refine existing KEP documentation
+- Define KEP [DACI]
+
+#### Organize
+
+- Move KEPs from flat-files to a directory structure:
+```
+├── 001-introduction.md (required)
+├── 002-motivation.md (required)
+├── 003-goals.md
+├── 004-non-goals.md
+├── 005-guide-for-developers.md (required)
+├── 006-guide-for-teachers.md (required)
+├── 007-guide-for-operators.md (required)
+├── 008-guide-for-project-maintainers.md
+├── 009-graduation-criteria.md (required)
+├── 010-implementation-history.md
+├── experience_reports (required)
+│   ├── alpha-feedback.md
+│   └── beta-feedback.md
+├── features
+│   ├── feature-01.md
+│   ├── feature-02.md
+│   └── feature-03.md
+├── index.json (required)
+├── metadata.yaml (required)
+└── OWNERS
+```
+
+metadata.yaml contains information that was previously in a KEP's YAML front-matter:
+
+```
+---
+authors: # required
+  - "calebamiles" # just a GitHub handle for now
+  - "jbeda"
+title: "Kubernetes Enhancement Proposal process"
+number: 42 # required
+owning-sig: "sig-pm" # required
+participating-sigs:
+  - "sig-architecture"
+  - "sig-contributor-experience"
+approvers: # required
+  - "bgrant0607" # just a GitHub handle for now
+reviewers:
+  - "justaugustus"  # just a GitHub handle for now
+  - "jdumars"
+editors:
+  - null # generally omit empty/null fields
+status: "active" # required
+github:
+  issues:
+    - null # GitHub url
+  pull_requests:
+    - null # GitHub url
+  projects:
+    - project_id: null
+      card_id: null
+releases: # required
+  - k8s_version: v1.9
+    kep_status: "active"
+    k8s_status: "alpha" # one of alpha|beta|GA
+  - k8s_version: v1.10
+    kep_status: "active"
+    k8s_status: "alpha"
+replaces:
+  - kep_location: null
+superseded-by:
+  - kep_location: null
+created: 2018-01-22 # in YYYY-MM-DD
+updated: 2018-09-04
+```
+
+- Move existing KEPs into [k/features]
+- Create a `kind/kep` label for [k/community] and [k/features]
+  - For `k/community`:
+    - Label incoming KEPs as `kind/kep`
+    - Enable searches of `org:kubernetes label:kind/kep`, so we can identify active PRs to `k/community` and reroute the PR authors to `k/keps` (depending on the state)
+  - For `k/keps` (fka `k/features`):
+    - Label incoming KEPs as `kind/kep`
+    - Classify KEP submissions / tracking issues as `kind/kep`, differentiating them from `kind/feature`
+- Move existing design proposals into [k/features]
+- Move existing architectural documents into [k/features]
+- Deprecate design proposals
+- Rename [k/features] to [k/keps]
+- Create tombstones / redirects to [k/keps]
+- Prevent new KEPs and design proposals from landing in [k/community]
+- Remove `kind/kep` from [k/community] once KEP migration is complete
+- Correlate existing features with KEPs
```

Comment:
Does this mean turning features into KEPs?

Reply:
We're introducing the idea of an umbrella term: `Enhancements`.
Enhancements are anything that modifies or adds new functionality to Kubernetes.

So KEPs and Features would both fall under that category.

KEP: multi-release event
Feature: hopefully only a single-release event

Features should roll into a KEP.

I think a good current example might be CoreDNS.
- Implementing CoreDNS == KEP
- CoreDNS as default in kubeadm == Feature

What I meant by this bullet point was that existing Feature issues should be associated to a KEP. We have a line for associating proposals to features in the `k/features` issue description, but not everyone fills it out. We'd want to backfill any missing links here.